app/javascript/components/server-components/ProductEditPage.tsx

```diff
@@ -150,38 +157,75 @@ const ProductEditPage = (props: Props) => {
 
   const [saving, setSaving] = React.useState(false);
   const [imagesUploading, setImagesUploading] = React.useState<Set<File>>(new Set());
+
+  const isBlocked =
+    saving ||
+    imagesUploading.size > 0 ||
+    product.files.some(
+      (file) =>
+        file.status.type === "unsaved" &&
+        file.status.uploadStatus.type === "uploading",
+    );
+
+  const [autosaveState, setAutosaveState] = React.useState<
+    "idle" | "saving" | "saved" | "error"
+  >("idle");
+
   const save = async () => {
```

Comment:
Refactor `save()` to support autosave semantics without changing existing save behavior.

This introduces autosave UI state (`autosaveState`), 
prevents overlapping saves, 
avoids alert spam during background saves,
and preserves existing content update logic and manual save feedback.